logstash-core/src/main/java/org/logstash/common/io/DeadLetterQueueWriter.java

```diff
@@ -96,14 +99,42 @@ private enum FinalizeWhen {ALWAYS, ONLY_IF_STALE};
     private ScheduledExecutorService flushScheduler;
     private final LongAdder droppedEvents = new LongAdder();
     private String lastError = "no errors";
+    private final Clock clock;
+    private Optional<Timestamp> oldestSegmentTimestamp;
+    private Optional<Path> oldestSegmentPath;
+    private final TemporalAmount retentionTime;
 
     public DeadLetterQueueWriter(final Path queuePath, final long maxSegmentSize, final long maxQueueSize,
                                  final Duration flushInterval) throws IOException {
         this(queuePath, maxSegmentSize, maxQueueSize, flushInterval, QueueStorageType.DROP_NEWER);
     }
 
+    public DeadLetterQueueWriter(final Path queuePath, final long maxSegmentSize, final long maxQueueSize,
```

Comment:
I've the doubt to introduce the builder because with constructors the client of this class knows immediately which are the parameters required, so it's clear at compile time. With builder we move this error at run time.
In the set of constructors we have 4 mandatory parameters (`queuePath`, `maxSegmentSize`, `maxQueueSize`, `flushInterval`) and 2 optional (`storageType` and `retentionTime`) plus a test only utility param `clock`.

I don't like the constructor with 7 parameters either, but at least it provides compile time safety.
WDYT?

Reply:
I like the suggestion, fixed 1a0f1c70faafa141dcc2ff85c18b280fd77d92b2